tools/autograd/templates/python_torch_functions_dispatch.h

```diff
@@ -28,6 +29,10 @@ static at::Type& default_type() {
   return *torch::tensors::get_default_tensor_type().aten_type();
 }
 
+static ScalarType default_scalar_type() {
+  return static_cast<ScalarType>(torch::tensors::get_default_tensor_type().scalar_type);
```

Comment:
Can we add a method to `torch//csrc/tensor/python_tensor.cpp` that does the cast for you? Doesn't have to be this PR, but the object here is a perfectly good class that can support nonvirtual methods.